### PR TITLE
Use an adhoc supervisor to start the dynamic supervisor and its initial children

### PR DESCRIPTION
**Pull request by adamzaninovich**
Creates an adhoc supervisor which will be started by the Application supervisor, start the dynamic supervisor, and then run a task that starts its children.

### CODE REVIEW

File: lib/in_stock/application.ex [hunk 1]
```diff
@@ -26,6 +25,33 @@ defmodule InStock.Application do
...
+  # supervisor CheckerSupervisor and then start its initial children
+  #
+  # References:
+  # - https://elixirforum.com/t/understanding-dynamicsupervisor-no-initial-children/14938/2
+  # - https://github.com/slashdotdash/til/blob/master/elixir/dynamic-supervisor-start-children.md
+  defp checker_child_spec() do
+    checker_children = [
+      InStock.Checker.CheckerSupervisor,
+      {Task, &InStock.Checker.CheckerSupervisor.spawn_checkers/0}
+    ]
```
Comment (adamzaninovich) [inline lib/in_stock/application.ex:38]: Sets up the children for the adhoc sup: the dynamic sup and then a task that runs its spawn_checkers

The task will run after the dynamic sup is started and ready

File: lib/in_stock/application.ex [hunk 2]
```diff
@@ -26,6 +25,33 @@ defmodule InStock.Application do
...
+
+    checker_opts = [
+      strategy: :rest_for_one,
+      name: InStock.Checker.CheckerStartSupervisor
+    ]
```
Comment (adamzaninovich) [inline lib/in_stock/application.ex:43]: rest for one strategy makes sure that the task cannot run or live unless the child before it (the dynamic sup) is up and running

File: lib/in_stock/application.ex [hunk 3]
```diff
@@ -26,6 +25,33 @@ defmodule InStock.Application do
...
+
+    %{
+      id: InStock.Checker.CheckerStartSupervisor,
+      start: {
+        Supervisor,
+        :start_link,
+        [checker_children, checker_opts]
+      }
+    }
```
Comment (adamzaninovich) [inline lib/in_stock/application.ex:52]: more verbose version of the child spec, just sets up the adhoc supervisor children and options

this gets passed to the application supervisor which will start every off

**Review by notdevinclark — APPROVED** [review]

**Pull request merged**